File: diagonal-difference.kt
fun diagonalDifference(arr: Array<Array<Int>>): Int {
    var topLeftToBottomRight: Int = 0
    var topRightToBottomLeft: Int = 0
    for (x in 0..arr.size - 1) {
        topLeftToBottomRight += arr[x][x]
        topRightToBottomLeft += arr[x][arr.size - (x + 1)]
    }
    return Math.abs(topLeftToBottomRight - topRightToBottomLeft)
}

File: diagonal-difference.js
function diagonalDifference(arr) {
  let primary = 0;
  let secondary = 0;
  arr.forEach((a, index) => {
      primary += a[index];
      secondary += a[a.length - (index + 1)];
  });
  return Math.abs(primary - secondary);
}